### PR TITLE
Fix checking existence of certain checks in the catalog

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

Fixes determination check existence in the catalog based on the target type or cluster type.

## How was this tested?

Automated tests.

### CODE REVIEW

File: assets/js/lib/model/checks.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+export const hasChecksForTarget = (catalog, targetType) =>
+  some(isCheckForTarget(targetType))(catalog);
+
+export const hasChecksForClusterType = (catalog, clusterType) =>
+  pipe(
+    filter(isCheckForTarget(TARGET_CLUSTER)),
+    some(
+      ({ metadata }) =>
+        !hasMetadata({ metadata }) ||
```
Comment (jamie-suse) [inline assets/js/lib/model/checks.js:21]: Isn't this check already done in the `filter()` step, and also later in `hasClusterTypeMetadata()`?

File: assets/js/lib/model/checks.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        !hasClusterTypeMetadata({ metadata }) ||
+        pipe(
```
Comment (jamie-suse) [inline assets/js/lib/model/checks.js:23]: Isn't this creating an array of size 1?

File: assets/js/lib/model/checks.js (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/lib/model/checks.js:21]: yes, but it is a different check.
`hasClusterTypeMetadata` checks whether there are metadata AND there is a `cluster_type` in it.

`!hasMetadata({ metadata }) ||` usage hear means that if there are not metadata we consider the check applicable for the given `clusterType`

File: assets/js/lib/model/checks.js (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/lib/model/checks.js:23]: you mean `castArray`?

If you're referring to it, it creates an array of size 1 if cluster_type is let's say a string like `ascs_ers`, but it covers cases where we have:
```yaml
metadata:
  target_type: cluster
  cluster_type:
    - hana_scale_up
    - hana_scale_out
```
maybe not frequent, but the metadata in the check work like this :smile:

File: assets/js/lib/model/checks.js (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/lib/model/checks.js]: nevermind, that seems to be safely removable 1408a32d91f6ba4b83ad3ccc95cabb38f4f7588f

File: assets/js/lib/model/checks.js (hunk 2 above)
Comment (jamie-suse) [reply assets/js/lib/model/checks.js:22]: Oh okay, so the type of `clusterType` is `String | [String]`?

Comment (nelsonkopliku) [reply assets/js/lib/model/checks.js:22]: potentially yes

**Review by jamie-suse — APPROVED** [review]

**Pull request merged**